### PR TITLE
Added browser URL for easier document access.

### PR DESCRIPTION
**Pull request by mhaehnel**
Adjusted configuration for this (!)

Completely untested. Would be nice if someone else could do this ...

### CODE REVIEW

File: detach.py [hunk 1]
```diff
@@ -349,7 +357,9 @@ def run(maildir, smtp_conn, exclude_seen, dir_pattern, learn_spam, learn_ham):
         exclude_seen = cfg.get("detach", "exclude-seen", fallback=True)
         smtp_host = cfg.get("smtp", "host", fallback="localhost")
         smtp_port = cfg.getint("smtp", "port", fallback=25)
-        dir_pattern = cfg.get("detach", "dir-pattern")
+        pattern = cfg.get("detach", "pattern")
+        dir_pattern = pattern+cfg.get("detach", "pattern")
+	url_pattern = pattern+cfg.get("detach", "pattern")
```
Comment (dirkonet) [inline detach.py]: I think you mixed up the configuration item names here.

File: detach.py [hunk 2]
```diff
@@ -40,6 +40,10 @@
 
     {destdir}
 
+the can be accessed via
+
```
Comment (dirkonet) [inline detach.py:44]: that/which?

File: README.rst [hunk 3]
```diff
@@ -22,10 +24,14 @@ is able to determine whether it is a moderation notice from mailman or not. The
...
 
-``dir-pattern`` specifies where attachments will be saved. It is first passed
+``pattern`` specifies where attachments will be saved. It is first passed
 through ``strftime(2)`` and then all occurences of ``{}`` are replaced by a
 name which is prompted from the user interactively.
 
+The actual directory is built by prefixing the result of the ``pattern`` 
+expansion with the ``dir`` setting. The mail also provides an URL where the
+attachments can be reached. The base URL is configured by the ``url`` setting
+and included in the mail with the expaneded ``pattern`` appended to it.
```
Comment (dirkonet) [inline README.rst]: expanded

File: detach.py (hunk 1 above)
Comment (mhaehnel) [reply detach.py]: hm ... that ammend did not come through somehow ... :S

File: detach.py (hunk 2 above)
Comment (mhaehnel) [reply detach.py:44]: they ;)

File: detach.py [hunk 4]
```diff
@@ -40,7 +40,7 @@
 
     {destdir}
 
-the can be accessed via
+they can be accessed via
```
Comment (dirkonet) [inline detach.py]: Would still vote for that or which here.

File: detach.py [hunk 5]
```diff
@@ -358,8 +358,8 @@ def run(maildir, smtp_conn, exclude_seen, dir_pattern, learn_spam, learn_ham):
         smtp_host = cfg.get("smtp", "host", fallback="localhost")
         smtp_port = cfg.getint("smtp", "port", fallback=25)
         pattern = cfg.get("detach", "pattern")
-        dir_pattern = pattern+cfg.get("detach", "pattern")
-	url_pattern = pattern+cfg.get("detach", "pattern")
+        dir_pattern = pattern+cfg.get("detach", "dir")
+	url_pattern = pattern+cfg.get("detach", "url")
```
Comment (dirkonet) [inline detach.py]: I think the sub strings should be switched - now it puts the folder structure before the url part.

File: detach.py [hunk 6]
```diff
@@ -114,13 +118,14 @@ def ask(prompt, options):
         print("Incorrect choice")
 
 
-def ask_nonexisting_dir(prompt, dirfmt):
+def ask_nonexisting_dir(prompt, dirfmt, urlfmt):
     while True:
         destdir = input(prompt)
         full = dirfmt.format(destdir)
+	fullurl = urlfmt.format(destdir)
```
Comment (dirkonet) [inline detach.py]: Please use spaces instead of tabs.

File: detach.py [hunk 7]
```diff
@@ -349,7 +357,9 @@ def run(maildir, smtp_conn, exclude_seen, dir_pattern, learn_spam, learn_ham):
         exclude_seen = cfg.get("detach", "exclude-seen", fallback=True)
         smtp_host = cfg.get("smtp", "host", fallback="localhost")
         smtp_port = cfg.getint("smtp", "port", fallback=25)
-        dir_pattern = cfg.get("detach", "dir-pattern")
+        pattern = cfg.get("detach", "pattern")
+        dir_pattern = cfg.get("detach", "dir")+pattern
+	url_pattern = cfg.get("detach", "url")+pattern
```
Comment (dirkonet) [inline detach.py]: Please use spaces instead of tabs.

File: detach.py [hunk 8]
```diff
@@ -374,6 +384,6 @@ def run(maildir, smtp_conn, exclude_seen, dir_pattern, learn_spam, learn_ham):
 
     conn = get_smtp_conn(smtp_host, smtp_port, args.verbose)
     try:
-        run(maildir, conn, exclude_seen, dir_pattern, learn_spam, learn_ham)
+        run(maildir, conn, exclude_seen, dir_pattern, url_pattern, learn_spam, learn_ham)
```
Comment (dirkonet) [inline detach.py:387]: TypeError: run() takes 6 positional arguments but 7 were given => Add url_pattern parameter to run() definition

**Comment by dirkonet** [comment]
Just tested this after applying the three last proposed changes: Works for me :)

**Comment by mhaehnel** [comment]
Updated as per dirks comments

**Pull request merged**